homeassistant/components/automation/zone.py

```diff
@@ -62,5 +70,8 @@ def zone_automation_listener(entity, from_s, to_s):
                 },
             }, context=to_s.context))
 
-    return async_track_state_change(hass, entity_id, zone_automation_listener,
-                                    MATCH_ALL, MATCH_ALL)
+    return async_track_state_change(hass, entity_id or entity_id_pattern,
+                                    zone_automation_listener,
+                                    MATCH_ALL, MATCH_ALL,
+                                    entity_id_with_pattern=bool(
```

Comment:
I don't think this is a good addition to the track state change helper, it's not common enough (yet).

Instead I suggest that you just add your own event listener for STATE_CHANGED_EVENTS inside this trigger.